editoast/src/views/timetable/paced_train.rs

```diff
@@ -748,6 +749,45 @@ mod tests {
         assert_eq!(response.len(), 1);
     }
 
+    #[rstest]
+    async fn update_paced_train_exception() {
+        let app = TestAppBuilder::default_app();
+        let pool = app.db_pool();
+
+        let timetable = create_timetable(&mut pool.get_ok()).await;
+        let simple_paced_train = simple_paced_train_changeset(timetable.id).exceptions(vec![]);
+        let mut simple_paced_train = simple_paced_train
+            .create(&mut pool.get_ok())
+            .await
+            .expect("Failed to create paced train");
+
+        assert_eq!(simple_paced_train.exceptions.len(), 0);
+
+        simple_paced_train.exceptions = vec![simple_created_exception_with_change_groups(
+            "exception_key_1",
+        )];
+        let paced_train: PacedTrain = simple_paced_train.clone().into();
+
+        let request = app
+            .put(format!("/paced_train/{}", simple_paced_train.id).as_str())
+            .json(&json!(&paced_train));
+
+        app.fetch(request).assert_status(StatusCode::NO_CONTENT);
+
+        #[expect(deprecated)]
```

Comment:
I think you should be able to use `retrieve_real` and remove the `#[expect(deprecated)]`.